sound/soc/codecs/rt1320-sdw.c

```diff
@@ -1488,17 +1486,6 @@ static bool rt1320_readable_register(struct device *dev, unsigned int reg)
 	case 0xf720 ... 0xf723:
 	case 0x1000f008:
 	case 0x3fe2e000 ... 0x3fe2e003:
```

Comment:
You could merge all these other 8-bit registers into the mbq_regmap as well, the drop that whole regmap.

Reply:
well, no, I do want a clean split between 
a) the 'SoundWire' registers (BRA possible, no deferrable action)
b) the 'SDCA' controls (BRA not allowed, deferrable action allowed)

If we have a single regmap it'll be a mess.

Reply:
:-) I guess perhaps that is where we were disagreeing a little through our other discussions around the regmap as well. I have definitely been leaning far more towards trying to have a single regmap if at all possible. More details on my thinking below, but if you are more comfortable with 2 register maps in this driver that is fine with me.

We added the deferrable callback so you can make only the registers you want deferrable, so that isn't really an issue, just don't mark the non-sdca register deferrable.

The BRA stuff this is perhaps more of a discussion, I haven't really worked through how best for that to all work yet. Although I would argue that since this part isn't actually using any BRA features the dangers of having multiple regmaps far outweights any issues around supporting BRA.

For example currently the driver calls rt1320_io_init from update_status, this function accesses a bunch of registers from rt1320_sdw_regmap. The driver also has a control "RX Channel Select" that accesses an MBQ from the rt1320_mbq_regmap. There isn't much to stop rt1320_io_init from accessing a register half way through the MBQ operation on the control, which would probably break the MBQ. The hw_init flag might maybe protect against this maybe user-space is always frozen whilst that flag is false, but it is kinda scary.

In general having multiple register maps is always going to involve a lot of very careful thinking around edge cases like that or some brute force locking in the driver across both regmaps.